File: src/WePrint/Models/PrinterModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.Reflection;

namespace WePrint.Models
{
    public class PrinterModel : DbModel
    {
        public string Name { get; set; }
        public PrinterType Type { get; set; }
        public double XMax { get; set; }
        public double YMax { get; set; }
        public double ZMax { get; set; }
        public double LayerMin { get; set; }

        public void ApplyChanges(PrinterUpdateModel update)
        {
            ReflectionHelper.CopyPropertiesTo(update, this);
        }
    }

    public class PrinterUpdateModel
    {
        public string Name { get; set; }
        public PrinterType? Type { get; set; }
        public double? XMax { get; set; }
        public double? YMax { get; set; }
        public double? ZMax { get; set; }
        public double? LayerMin { get; set; }
    }
}

File: src/WePrint/ClientApp/src/models/PrinterModel.js

export default class PrinterModel {
	constructor() {
		this.id = "";
		this.ownerId = "";
		this.name = "";
		this.type = "";
		this.xMax = "";
		this.yMax = "";
		this.zMax = "";
		this.layerMin = "";
	}

	static AllPropsEqual(a, b) {
		return (
			a.id === b.id &&
			a.ownerId === b.ownerId &&
			a.name === b.name &&
			a.type === b.type &&
			a.xMax === b.xMax &&
			a.yMax === b.yMax &&
			a.zMax === b.zMax &&
			a.layerMin === b.layerMin
		);
	}
}
